src/textual/app.py

```diff
@@ -2059,7 +2061,8 @@ async def invoke_ready_callback() -> None:
                     Reactive._initialize_object(self)
 
                     self.stylesheet.update(self)
-                    self.refresh()
+                    if self.screen is not default_screen:
+                        self.stylesheet.update(default_screen)
```

Comment:
What can happen in between the time where you set `default_screen = self.screen` and this `if` statement that `self.screen` is no longer what you stored inside `default_screen`?

Reply:
The issue was that if you change the screen stack in App.on_mount then the default screen never has its CSS applied.

Reply:
Ahhhh, I see. Thanks.